50_wa_diff_overdose.py

```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import altair as alt
+from plotnine import *
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+
+# load in pre-cleaned deaths data for Washington and comparison states
+deaths = pd.read_csv("05_cleaned_data/deaths_wa.csv")
+# two separate dfs - one for washington and one for comp states
+washington = deaths[deaths["StateName"] == "Washington"]
+# comp = deaths[deaths["StateName"] != "Washington"]
+comp = deaths[
+    (deaths["StateName"] == "Oregon")
+    | (deaths["StateName"] == "Idaho")
+    | (deaths["StateName"] == "Montana")
+    | (deaths["StateName"] == "Nevada")
+    | (deaths["StateName"] == "Wyoming")
+]
+
+
+def select_overdose(record):
+    """Simple function to select only overdose records"""
+
+    if record == "All other non-drug and non-alcohol causes":
+        return 0
+
+    if record == "All other alcohol-induced causes":
+        return 0
+
+    if record == "All other drug-induced causes":
+        return 0
+
+    if record == "Alcohol poisonings (overdose) (X45, X65, Y15)":
+        return 0
+
+    if record == "Drug poisonings (overdose) Unintentional (X40-X44)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Suicide (X60-X64)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Undetermined (Y10-Y14)":
+        return 1
+
+    else:
+        return "error"
```

Comment:
This helper function would work fine. But there is another way, you could have done it -

Creating the dataset for Washington which has all deaths that are drug related

wa_deaths = deaths[
(deaths["State"] == "Washington")
& (
(
deaths["Drug/Alcohol Induced Cause"]
== "Drug poisonings (overdose) Unintentional (X40-X44)"
)
| (
deaths["Drug/Alcohol Induced Cause"]
== "Drug poisonings (overdose) Undetermined (Y10-Y14)"
)
| (
deaths["Drug/Alcohol Induced Cause"]
== "Drug poisonings (overdose) Suicide (X60-X64)"
)
)
]